src/PowerShellEditorServices/Server/PsesLanguageServer.cs

```diff
@@ -149,13 +144,18 @@ public async Task StartAsync()
                                 //
                                 // NOTE: The keys start with a lowercase because OmniSharp's client
                                 // (used for testing) forces it to be that way.
-                                LoadProfiles = initializationOptions?.GetValue("enableProfileLoading")?.Value<bool>() ?? true,
-                                // TODO: Consider deprecating the setting which sets this and
-                                // instead use WorkspacePath exclusively.
-                                InitialWorkingDirectory = initializationOptions?.GetValue("initialWorkingDirectory")?.Value<string>() ?? workspaceService.WorkspacePath,
-                                ShellIntegrationEnabled = initializationOptions?.GetValue("shellIntegrationEnabled")?.Value<bool>() ?? false
+                                LoadProfiles = initializationOptions?.GetValue("enableProfileLoading")?.Value<bool>()
+                                    ?? true,
+                                InitialWorkingDirectory = initializationOptions?.GetValue("initialWorkingDirectory")?.Value<string>()
+                                    ?? workspaceService.WorkspaceFolders.FirstOrDefault()?.Uri.GetFileSystemPath(),
+                                ShellIntegrationEnabled = initializationOptions?.GetValue("shellIntegrationEnabled")?.Value<bool>()
+                                    ?? false
                             };
 
+                            // NOTE: We're still maintaining a root path while things transition to
+                            // support multi-root workspaces.
+                            workspaceService.WorkspacePath = hostStartOptions.InitialWorkingDirectory;
+
```

Comment:
The main thing we still use this for is resolving the path to the PSScriptAnalyzer settings file. People give relative paths...but we have no way of asking "ok, but relative to which root?" and so for now the logic is "your initial working directory setting" first (so it can be corrected if necessary) and then fall back to the "first" (but presumably random sorting) workspace folder.